### PR TITLE
#2682 the sum of name and group length should less or equal than 100 in clusters

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #2682)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala [hunk 1]
```diff
@@ -300,6 +319,17 @@ trait JsonRefiner[T] {
...
+  /**
+    * add your custom check for a set of keys' values.
+    * Noted: We don't guarantee the order of each (key, value) pair since this method is intend
+    * to do an "aggregation" check, like sum or length check.
+    *
+    * @param keys keys
+    * @param checkers checkers
+    * @return this refiner
+    */
+  def allValuesChecker(keys: Set[String], checkers: Set[JsValue] => Unit): JsonRefiner[T]
```
Comment (chia7712) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala]: what happen if the keys of input request don't match the "keys" completely?

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala [hunk 2]
```diff
@@ -615,6 +651,11 @@ object JsonRefiner {
           // 5) check the keys
           keyCheckers.foreach(_(fields.keySet))
 
+          // 6) check the values of specific keys
+          allvaluesCheckers.foreach {
+            case (keys, f) => f(fields.filter(field => keys.contains(field._1)).values.toSet)
```
Comment (chia7712) [inline ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala]: toSet de-duplicates the values. For example:

```json
{
  "name": "a",
  "group": "a"
}
```

the stuff passed via allvaluesCheckers will be "a" than "a", "a"

File: ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala (hunk 1 above)
Comment (saivirtue) [reply ohara-client/src/main/scala/com/island/ohara/client/configurator/v0/JsonRefiner.scala]: Neglect in my first design. But requireKeys should be more suitable.

**Comment by konekoya** [comment]
retry

**Comment by saivirtue** [comment]
retry configurator

**Pull request merged**